tests/sles4sap/sap_cluster.pm

```diff
@@ -0,0 +1,162 @@
+    my $sid          = get_required_var('INSTANCE_SID');
+    my $alias        = get_var('INSTANCE_ALIAS');
+    my $lun          = get_var('INSTANCE_LUN');
+    my $cluster_name = get_cluster_name;
+    my ($ip, $netmask) = split '/', get_var('INSTANCE_IP');
+
+    # Set SAP variables
+    my $pscmd  = $self->set_ps_cmd("$type");
+    my $sapadm = $self->set_sap_info($sid, $instance_id);
+
+    # Synchronize the nodes
+    barrier_wait "SAP_CLUSTER_INSTALL_$cluster_name";
+
+    $self->select_serial_terminal;
+
+    if (get_var('NW')) {
+        # Stop the SAP instance to add it in the cluster stack
+        $self->user_change;
+        $self->test_version_info;
+        $self->test_instance_properties;
+        $self->test_stop;
+
+        # Disconnect SAP account
+        $self->reset_user_change;
+
+        # Some file changes are needed for HA
+        $self->select_serial_terminal;
+
+        my $profile_file = "/usr/sap/$sid/SYS/profile/${sid}_${type}${instance_id}_${alias}";
+
+        assert_script_run 'cp /sapinst/sapservices /usr/sap/';
+        assert_script_run "echo 'service/halib = \${DIR_CT_RUN}/saphascriptco.so' >> $profile_file";
+        assert_script_run "echo 'service/halib_cluster_connector = /usr/bin/sap_suse_cluster_connector' >> $profile_file";
+        assert_script_run "sed -i 's/^Restart_Program_\\(.*[[:blank:]]*=\\)/Start_Program_\\1/' $profile_file";
+
+        # Add SAP account into haclient group
+        assert_script_run "usermod -a -G haclient $sapadm";
+
+        # Removed uneeded(?) profile directory for ERS
+        if ($type eq 'ERS') {
+            assert_script_run "rm -rf /usr/sap/$sid/${type}${instance_id}/profile";
+            assert_script_run "ln -s /sapmnt/$sid/profile /usr/sap/$sid/${type}${instance_id}/profile";
+            assert_script_run "chown -h $sapadm:sapsys /usr/sap/$sid/${type}${instance_id}/profile";
+        }
+    }
+
+    # Create the resource configuration
+    my @sedoptions = undef;
+    push @sedoptions, "-e 's|%SID%|$sid|g'";
+    push @sedoptions, "-e 's|%${type}_INSTANCE%|$instance_id|g'";
+    push @sedoptions, "-e 's|%${type}_ALIAS%|$alias|g'";
+    push @sedoptions, "-e 's|%${type}_LUN%|$lun|g'";
+    push @sedoptions, "-e 's|%${type}_IP%|$ip|g'";
+    if (check_var('CLUSTER_NAME', 'hana')) {
+        my $automated_register = get_required_var('AUTOMATED_REGISTER');
+        my ($virtual_ip, $virtual_netmask) = split '/', get_required_var('VIRTUAL_IP_CIDR');
+        push @sedoptions, "-e 's|%AUTOMATED_REGISTER%|$automated_register|g'";
+        push @sedoptions, "-e 's|%VIRTUAL_IP_ADDRESS%|$virtual_ip|g'";
+        push @sedoptions, "-e 's|%VIRTUAL_IP_NETMASK%|$virtual_netmask|g'";
+    }
+
+    # We need to execute the sed command node by node
+    my $cmd          = undef;
+    my $cluster_conf = undef;
+    if (get_var('NW')) {
+        $cluster_conf = 'nw_cluster.conf';
+    } else {
+        $cluster_conf = 'hana_cluster.conf';
```

Comment:
Maybe better to condition this to `get_var('HANA')`? Should do the same but maybe more readable.

Reply:
What about:

```
my $cluster_conf = get_var('NW') ? 'nw_cluster.conf' : 'hana_cluster.conf';
my $cmd = 'sed -i' . join(' ', @sedoptions) . " $cluster_conf";
```

There's also the method `file_content_replace` in `lib/utils` which could help to clear some of the code.